### ISSUE TITLE
Scenario Results: Assurance of threshold

### ISSUE THREAD

**Issue by codymccoy** [HydrologicEngineeringCenter/HEC-FDA#402]
All of my results are NaN:
![image](https://user-images.githubusercontent.com/47757848/174100249-be677498-b92d-42e2-8760-39c43d6dc170.png)

This is the call to get the values:
![image](https://user-images.githubusercontent.com/47757848/174101314-0ceb3686-93eb-4e95-bc79-526f4f7cdf71.png)

In this case i don't have any extra thresholds, only the default. The threshold key is zero.

**Comment by rnugent3** [comment]
using default data?

**Comment by codymccoy** [comment]
Yes, this was using the default curves for rating and stage damage.

**Comment by rnugent3** [comment]
I am successfully getting assurance results for a handful of different study data configuration possibilities and using default data. 

I made the default value for the threshold ID = 0. The following calls all return numbers:

``` 
            double actualCNEP1 = scenarioResults.AssuranceOfEvent(impactAreaID, .9);
            double actualCNEP04 = scenarioResults.AssuranceOfEvent(impactAreaID, .96);
            double actualCNEP02 = scenarioResults.AssuranceOfEvent(impactAreaID, .98);
            double actualCNEP01 = scenarioResults.AssuranceOfEvent(impactAreaID, .99);
            double actualCNEP004 = scenarioResults.AssuranceOfEvent(impactAreaID, .996);
            double actualCNEP002 = scenarioResults.AssuranceOfEvent(impactAreaID, .998);

**Comment by rnugent3** [comment]
Also - looks like your probabilities don't match exactly. I know the hard-coding is terrible, I have it on my to do list to get rid of the hard-coded everything 

The standard non-exceedance probabilities are:
```
          double[] er101RequiredNonExceedanceProbabilities = new double[] { .9, .96, .98, .99, .996, .998 };

**Comment by rnugent3** [comment]
at the simulation level, passing a key in works. it could be an issue farther up the food chain.
